batch/batch/front_end/front_end.py

```diff
@@ -432,6 +528,15 @@
 
                 id = (batch_id, job_id)
 
+                if start_job_id is None:
```

Comment:
I can't comment there, but ~5 lines above, do you want to pop the attributes in v2 or do you want to store them in the spec-in-gs?  I don't think it really matters but this saves a bit.